### DISCUSSION TITLE
%topic% not resolved in scripts when using tasmota_%06X as topic

### DISCUSSION THREAD

**Discussion by Rainerlan** [arendst/Tasmota#16809] [Q&A]
Hi, 
Can anyone tell, 
if the following behavior is 
<img width="264" alt="Screenshot 2022-10-13 182310" src="https://user-images.githubusercontent.com/179617/195652171-9c2dfcf3-c6ec-45b7-84d2-494de7ec3dc2.png">
only a problem in 11.1.0 - or if this does still exist in the current release?
(or if its maybe a known limitation of the scripting language - at least of the ">F"-section)

**Problem:**
Using the %topic% in a script in the ">F" section...:
`=>Publish tele/%topic%/SENSOR {"Script":{"button":%status%}}`

...the 3bytes from the MAC are not correctly replaced.
Instead, the following is being published:
`17:03:26.866 MQT: tele/tasmota_%06X/SENSOR = {"Script":{"button":1.00}}`

The publish of "other data" - e.g. decvice state or other sensors - is using the correct topic:
`17:15:45.543 MQT: tele/tasmota_0168C2/STATE = {"Time":"2022-10-13T17:15:45","Uptime":"0T00:25:14", ...`

Find attached a screenshot of the setup of the "topic" from the configuration.

**Comment by barbudor** [comment]
I see what seems to be wrong
Are you compiling yourself ? I would assume yes. How ? TasmoCompiler or VSCode ?

**Reply by barbudor** [reply]
Fixed with https://github.com/arendst/Tasmota/pull/16810
Please wait for PR to be merged

**Reply by Rainerlan** [reply]
Thanks a lot!
To answer your question: I used a binary someone else had built - but when I compile the project, I use PlatformIO in VSC.
Once the PR is merged, I'll need to compile myself anyway.
